controllers/storageautoscaler/scraper_test.go

```diff
@@ -1 +1,48 @@
 package storageautoscaler
+
+import (
+	"sync"
+	"testing"
+
+	"github.com/go-logr/logr"
+	ocsv1 "github.com/red-hat-storage/ocs-operator/api/v4/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func TestFilterObjectsForScaling(t *testing.T) {
+	scalerList := ocsv1.StorageAutoScalerList{
+		Items: []ocsv1.StorageAutoScaler{
+			{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "ssdscaler",
+				},
+				Spec: ocsv1.StorageAutoScalerSpec{
+					DeviceClass:                    "deviceClass",
+					StorageScalingThresholdPercent: 50,
+				},
+			},
+		},
+	}
+
+	syncMap := &sync.Map{}
+	syncMap.Store("deviceClass", 0.60)
+
+	log := logr.Logger{}
+	objectList := filterObjectsForScaling(&scalerList, syncMap, log)
+
+	if len(objectList) != 1 {
+		t.Errorf("Expected 1 object in the list, got %v", len(objectList))
+	}
+
+	if objectList[0].name != "ssdscaler" {
+		t.Errorf("Expected object name to be deviceClass, got %v", objectList[0].name)
+	}
+
+	syncMap.Store("deviceClass", 0.40)
+
+	objectList = filterObjectsForScaling(&scalerList, syncMap, log)
+
+	if len(objectList) != 0 {
```

Comment:
Several places in this test above and here can use `assert.NotEqual()`, for example:
`assert.NotEqual(t, 0, len(objectList))`